core/shared/src/main/scala/fs2/async/Promise.scala

```diff
@@ -29,48 +25,37 @@ import Promise._
   *
   * Finally, the blocking mentioned above is semantic only, no actual threads are blocked by the implementation.
   */
-final class Promise[F[_], A] private[fs2] (ref: Ref[F, State[A]])(implicit F: Effect[F],
-                                                                  ec: ExecutionContext) {
+final class Promise[F[_], A] private[fs2] (ref: AtomicReference[State[A]])(
+    implicit F: Concurrent[F]) {
```

Comment:
Since the ExecutionContext is not needed here anymore, can this change ripple up and we remove it pretty much everywhere? (not saying it should be done in this PR, but I want to know if it's possible).

I've also almost entirely convinced myself that removing it here is good, and that the callbacks don't have to be executed on the Promise ec, but I'd like to hear a few words motivating this change to have full peace of mind, if possible :)

Reply:
I tried heading down that path but wanted to get your opinion. Specifically, `Signal` still requires an `ec` in order to fork the completion of the promise (and hence, fork the callback invocations that are blocked on the promise's `get`).

Reply:
Right, in that case I'd say this deserves its own exploration/issue/PR.
Just wanted to make sure I wasn't missing something obvious that would make it a clear decision one way or another.